code/modules/reagents/machinery/chem_master.dm

```diff
@@ -446,49 +445,76 @@ var/global/list/pillIcon2Name = list("oblong purple-pink", "oblong green-white",
 			return 1
 
 		else if (href_list["createbottle"] || href_list["createbottle_multiple"])
+			var/count = 1
 			if(reagents.total_volume == 0)
 				to_chat(usr, "<span class='warning'>[bicon(src)] Buffer is empty!</span>")
 				return
+			var/amount_per_bottle = reagents.total_volume > 0 ? reagents.total_volume/count : 0
+			amount_per_bottle = min(amount_per_bottle,max_bottle_size)
+			
 			if(!condi)
-				var/count = 1
 				if(href_list["createbottle_multiple"])
 					count = isgoodnumber(input("Select the number of bottles to make.", "Amount:", last_bottle_amt) as num)
 				count = clamp(count, 1, 4)
 				last_bottle_amt = count
 
-				var/amount_per_bottle = reagents.total_volume > 0 ? reagents.total_volume/count : 0
-				amount_per_bottle = min(amount_per_bottle,max_bottle_size)
+			var/name = stripped_input(usr,"Name:", "Name your bottle!","[reagents.get_master_reagent_name()] ([amount_per_bottle] units)")
+			if(!name)
+				to_chat(usr, "<span class='warning'>[bicon(src)] Invalid name!</span>")
+				return
 
-				var/name = stripped_input(usr,"Name:", "Name your bottle!","[reagents.get_master_reagent_name()] ([amount_per_bottle] units)")
-				if(!name)
-					to_chat(usr, "<span class='warning'>[bicon(src)] Invalid name!</span>")
-					return
-
-				while(count--)
-					if(amount_per_bottle == 0 || reagents.total_volume == 0)
-						break
-
-					var/obj/item/weapon/reagent_containers/glass/bottle/unrecyclable/P = new/obj/item/weapon/reagent_containers/glass/bottle/unrecyclable/(src.loc,max_bottle_size)
-					P.name = "[name] bottle"
-					P.pixel_x = rand(-7, 7) * PIXEL_MULTIPLIER//random position
-					P.pixel_y = rand(-7, 7) * PIXEL_MULTIPLIER
-					//P.icon_state = "bottle"+bottlesprite
-					reagents.trans_to(P,amount_per_bottle)
-				src.updateUsrDialog()
-				return 1
-			else
-				var/obj/item/weapon/reagent_containers/food/condiment/P = new/obj/item/weapon/reagent_containers/food/condiment(src.loc)
-				reagents.trans_to(P, 50)
-				src.updateUsrDialog()
-				return 1
+			while(count--)
```

Comment:
while count>=0 is safer because if count somehow broke and became -1, this would become an infinite loop